File: euler003/solution.cpp
//Largest prime factor
bool prime(long long d){
	if(d==2)return true;
	else if(d%2==0) return false;
	else {
		for(int i=3;i<=floor(sqrt((double)d)) ;i+=2){
			if(d%i==0)
				return false;
		}
		return true;
	}
}

int _tmain(int argc, _TCHAR* argv[])
{
	//int num=210;
	long long num=600851475143 ;
	for(int i=3;;i+=2){
		printf(""i=%d num=%d \n"", i,num);
		while(num%i==0 && prime(i))
			num=num/i;
		if(prime(num))break;
	}
	printf(""Largest prime factor is %d \n"", num);
	return 0;
}

File: euler003/solution.py
#!/usr/bin/python
# Largest prime factor

import sys
sys.path.append("../common_py")
from prime import *

def main():
	i=3
	N=600851475143
	while True:
		if N%i==0 and isPrime(i):
			N=N/i
			print N
			if isPrime(N):
				break
		i=i+2
	print N



main()	